main.js

```diff
@@ -0,0 +1,118 @@
+// selecteren van alle items op de pagina
+const searchForm = document.querySelector(".form");
+const searchButton = document.querySelector(".form button");
+const searchInput = document.querySelector(".form input");
+
+let landContainer = document.querySelector(".land-container");
+let landHeader = document.querySelector(".line-header");
+let landList = document.querySelector(".line-content");
+
+// aanmaken van DOM elementen
+let landImg = document.createElement("img");
+let landH2 = document.createElement("h2");
+let firstline = document.createElement("li");
+let secondline = document.createElement("li");
+let thirdline = document.createElement("li");
+
+// Functie om de data uit de API te halen
+// prettier-ignore
+const fetchData = async function (search) {
+  try {
+    const result = await axios.get(`https://restcountries.eu/rest/v2/name/${search}`);
+    return result;
+
+  } catch (e) {
+    cleanElements();
+    console.log("ERROR");
+    alert("ERROR, CAN'T FIND ANY RESULTS");
+    console.error(e);
+  }
+};
+
+// Functie voor het printen van meerdere talen
+// prettier-ignore
+const printLanguage = function (languages) {
+  
+  const firstLanguages = languages
+    .map((e) => e.name)
+    .slice(0, -1)
+    .join(", ");
+
+  const lastLangauge = languages
+    .map((e) => e.name)
+    .slice(-1);
+
+  if (languages.length > 1) {
+    return `They speak ${firstLanguages} and ${lastLangauge}`;
+  } else {
+    return `They speak ${lastLangauge}`;
+  }
+};
+
+//Functie voor het deleten van DOM elementen
+const cleanElements = function () {
+  landContainer.removeAttribute("class", "land-container");
+  landHeader.removeAttribute("class", "line-header");
+  landList.removeAttribute("class", "line-content");
+  landImg.src = "";
+  landH2.textContent = "";
+  firstline.textContent = "";
+  secondline.textContent = "";
+  thirdline.textContent = "";
+};
+
+// Functie voor het zetten van classes
+const setClasses = function () {
+  landContainer.setAttribute("class", "land-container");
+  landHeader.setAttribute("class", "line-header");
+  landList.setAttribute("class", "line-content");
+};
+
+//Functie print alle data uit (main functie) prettier-ignore
+// prettier-ignore
+const printData = async function (search) {
+  const result = await fetchData(search);
```

Comment:
Wat ik zelf heel vervelend vind is dat ik heel veel `result.data[0]` moet herhalen. Wij developers zijn heel lui, dus is daarom de object spreader in het leven geroepen 😉 
```
const { name: countryName, population, subregion: sbuAreaName, ...otherProps } = result.data[0];
```

_Je hoeft de props niet perse te aliasen_

Reply:
thx, zeker goeie aanvulling. Het hele deconstructen en gebruik vd spread operator zit er nog niet helemaal in.